packages/devtools_app/lib/src/eval_on_dart_library.dart

```diff
@@ -156,25 +178,229 @@ class EvalOnDartLibrary {
     }
   }
 
-  Future<Library> getLibrary(LibraryRef instance, ObjectGroup isAlive) {
+  Future<Library> getLibrary(LibraryRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
-  Future<Class> getClass(ClassRef instance, ObjectGroup isAlive) {
+  Future<Class> getClass(ClassRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
-  Future<Func> getFunc(FuncRef instance, ObjectGroup isAlive) {
+  Future<Func> getFunc(FuncRef instance, Disposable isAlive) {
     return getObjHelper(instance, isAlive);
   }
 
   Future<Instance> getInstance(
     FutureOr<InstanceRef> instanceRefFuture,
-    ObjectGroup isAlive,
+    Disposable isAlive,
   ) async {
     return await getObjHelper(await instanceRefFuture, isAlive);
   }
 
+  Future<int> getHashCode(
+    InstanceRef instance, {
+    @required Disposable isAlive,
+  }) async {
+    final hash = await evalInstance(
+      'instance.hashCode',
+      isAlive: isAlive,
+      scope: {'instance': instance.id},
+    );
+
+    return int.parse(hash.valueAsString);
+  }
+
+  /// Eval an expression and immediately obtain its [Instance].
+  Future<Instance> evalInstance(
+    String expression, {
+    @required Disposable isAlive,
+    Map<String, String> scope,
+  }) async {
+    final ref = await safeEval(expression, isAlive: isAlive, scope: scope);
+    if (ref == null) return null;
+
+    return getInstance(ref, isAlive);
+  }
+
+  static int _nextAsyncEvalId = 0;
+
+  EvalOnDartLibrary _dartDeveloperEvalCache;
```

Comment:
you should dispose these caches if they are non null in the dispose method for this eval.